aries_cloudagent/core/plugin_registry.py

```diff
@@ -193,10 +193,12 @@ def register_package(self, package_name: str) -> Sequence[ModuleType]:
             )
         )
 
-    async def init_context(self, context: InjectionContext):
+    async def init_context(self, context: InjectionContext, plugins_config: dict = {}):
         """Call plugin setup methods on the current context."""
-        for plugin in self._plugins.values():
+        for key, plugin in self._plugins.items():
             if hasattr(plugin, "setup"):
+                plugin_conf = plugins_config.get(key)
+                context.update_settings({"plugin_conf": plugin_conf})
```

Comment:
I don't think these changes are necessary either. The modified settings in `argparse.py` should already be present on the context as `context.settings["plugin_config"]` (assuming `plugins_config` -> `plugin_config` change is made).

Reply:
Finally I decided to remove the `plugins_config` because we can consider some "sensitive information" like users/passwords, or whatever that should not be propagated and should be confidential for each plugin.